app/scripts/metamask-controller.js

```diff
@@ -1507,10 +1506,39 @@ export default class MetamaskController extends EventEmitter {
       signatureController: this.signatureController,
       platform: this.platform,
       extension: this.extension,
-      trackTransactionEvents: handleMMITransactionUpdate.bind(
-        null,
-        transactionMetricsRequest,
-      ),
+      hooks: {
+        getTransactions: this.txController.getTransactions.bind(
+          this.txController,
+        ),
+        updateTransactionHash: this.txController.setTxHash.bind(
+          this.txController,
+        ),
+        trackTransactionEvents: handleMMITransactionUpdate.bind(
+          null,
+          transactionMetricsRequest,
+        ),
+        txStateManager: {
+          getTransactions: this.txController.getTransactions.bind(
+            this.txController,
+          ),
+          setTxStatusSigned:
+            this.txController.txStateManager.setTxStatusSigned.bind(
+              this.txController.txStateManager,
+            ),
+          setTxStatusSubmitted:
+            this.txController.txStateManager.setTxStatusSubmitted.bind(
+              this.txController.txStateManager,
+            ),
+          setTxStatusFailed:
+            this.txController.txStateManager.setTxStatusFailed.bind(
+              this.txController.txStateManager,
+            ),
+          updateTransaction:
+            this.txController.txStateManager.updateTransaction.bind(
```

Comment:
Actually this `updateTransaction` can handle all the updates, not sure why they added all other updates methods before.

Reply:
As we're trying to move away from the generalised update method which will eventually become `private`, shall we instead check what is specifically being updated and create a suitable `updateCustodialTransaction` method in core?

Reply:
100%! Added in the core a `updateCustodialTransaction` from what I could check apart from `status` and `hash` the other property updated  using `updateTransaction` method is the `custotyStatus`. I have also included it in the custodian method in core.